Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/UserMetadataService.java

```diff
@@ -58,6 +60,15 @@ void saveUserMetadataLinkedToOneLog(String userMetadataContent, UserMetadataType
                                         String username,
                                         Integer logId) throws UserNotFoundException;
 
+    /**
+     * Save a user metadata which is not linked to log. So it can not be shared to other users at stage one.
+     *
+     * @param content Content of user metadata
+     * @param username username
+     * @throws UserNotFoundException Can't find a user with specified username
+     */
+    void saveUserMetadataWithoutLog(String content, UserMetadataTypeEnum userMetadataTypeEnum, String username) throws UserNotFoundException;
```

Comment:
Keeping both UserMetadataType (values stored in database) and UserMetadataTypeEnum (values hard-coded) would create issues later if the hard-coded values in UserMetadataTypeEnum are inadvertently changed and out of sync with the values stored for UserMetadataType in the database. We should use consistently UserMetadataType in everywhere in the code if it needs to be used, no need another duplication like UserMetadataTypeEnum.

Reply:
Suggest to replace UserMetadataTypeEnum param used in many methods with userMetadataTypeId (integer type).

Reply:
Create a separate card for this refactoring.
https://apromore.atlassian.net/browse/AP-2454